volume/volume.go

```diff
@@ -79,8 +76,7 @@ func (m *MountPoint) Setup() (string, error) {
 				return "", err
 			}
 			if runtime.GOOS != "windows" { // Windows does not have deprecation issues here
-				logrus.Warnf("Auto-creating non-existent volume host path %s, this is deprecated and will be removed soon", m.Source)
-				if err := system.MkdirAll(m.Source, 0755); err != nil {
+				if err := os.MkdirAll(m.Source, 0755); err != nil {
```

Comment:
Yes, if we decide to go for https://github.com/docker/docker/pull/22065#issuecomment-210494250, it has to be changed back